opent2t.p.light/com.wink.lightbulb/js/thingTranslator.js

```diff
@@ -0,0 +1,207 @@
+/* jshint esversion: 6 */
+/* jshint node: true */
+'use strict';
+
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+// Wink does not always populate every desired_state property, but last_reading doesn't necessarily
+// update as soon as we send our PUT request. Instead of relying just on one state or the other,
+// we use this StateReader class to read from desired_state if it is there, and fall back to last_reading
+// if it is not.
+class StateReader {
+    constructor(desired_state, last_reading) {
+        this.desired_state = desired_state;
+        this.last_reading = last_reading;
+    }
+
+    get(state) {
+        if (this.desired_state[state] !== undefined) {
+            return this.desired_state[state];
+        }
+        else {
+            return this.last_reading[state];
+        }
+    }
+}
+
+// Helper method to convert Wink's 0.0-1.0 brightness scale to a 0-100 scale
+function scaleDeviceBrightnessToTranslatorBrightness(brightnessValue) {
+    return Math.floor(brightnessValue * 100);
+}
+
+// Helper method to convert a 0-100 scale to Wink's 0.0-1.0 brightness scale
+function scaleTranslatorBrightnessToDeviceBrightness(dimmingValue) {
+    return dimmingValue / 100;
+}
+
+/**
+ * Finds a resource for an entity in a schema
+ */
+function findResource(schema, di, resourceId) {
+    // Find the entity by the unique di
+    var entity = schema.entities.find((d) => {
+        return d.di === di;
+    });
+
+    // Find the resource
+    var resource = entity.resources.find((r) => {
+        return r.id === resourceId;
+    });
+
+    return resource;
+}
+
+/***
+ * 
+ */
+function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
+    // build the object with desired state
+    var result = { 'desired_state': {} };
+    var desired_state = result.desired_state;
+
+    switch(resourceId) {
+        case 'power':
+            desired_state['powered'] = resourceSchema.value;
+            break;
+        case 'dim':
+            desired_state['brightness'] = scaleTranslatorBrightnessToDeviceBrightness(resourceSchema.dimmingSetting);
+            break;
+        default:
+            // Error case
+            console.log("oops");
+    }
+
+    return result;
+}
+
+/**
+ * 
+ */
+function providerSchemaToPlatformSchema(providerSchema, expand) {
+    var stateReader = new StateReader(providerSchema.desired_state, providerSchema.last_reading);
+
+    // Build the oic.r.switch.binary resource
+    var binarySwitch = {
```

Comment:
You may want to consider naming these variables after the name of the resource instead of the type of the resource, otherwise the precedent gets hard to follow when we have multiple resources of the same type that we need to deal with.